pkg/server/testserver_sqlconn.go

```diff
@@ -53,13 +56,19 @@ func openTestSQLConn(
 	cleanupFn := func() {}
 	var goDB *gosql.DB
 
+	opts := url.Values{}
+	if tenantName != "" {
+		opts.Add("options", fmt.Sprintf("-ccluster=%s", tenantName))
+	}
+
```

Comment:
I could see lifting sslmode up here behind something like `if useLoopbackListner || insecure {`, but I leave that style choice to you.